Import/Export: Display progress only if not syncing

epeicher opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-465

## Proposed Changes

- Refine the condition to display the Export progress bar so it is not displayed when Syncing
- Additionally, do not display the Export progress bar when it reaches 100 as it will be disabled anyway.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Run `npm start`
3. Go to Sync tab
5. Connect a site
7. Click on Push
9. Observe the progress bar appears with messages
11. Go to Import/Export tab
13. The progress bar should not be visible
15. When the Push finishes, the export progress bar should not be visible and the button should be enabled

In the linked task, there is a Video with the expected behaviour, [link here](https://linear.app/a8c/issue/STU-465/when-pushing-a-site-a-progress-bar-appears-in-the-importexport-tab#heading-expectation-9c56de4f)

#### Regression testing
11. Go to Import/Export tab
12. Export a site
13. The progress bar should be work as before. It should be displayed and progressed.
15. When the Export finishes, the export progress bar should not be visible and the button should be enabled


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

epeicher committed:
Import/Export: Display progress if not syncing

sejas reviewed (commented):
It seems to solve the issue, but I have found another edge case where it fails.

1. Push the site.
2. Wait for the push to finish
3. Got to Import/Export tab and Try exporting the site
4. Observe the dialog opens but the export doesn't run for that site. The export works for the rest of sites.

See the vide at 1min 19seg.

https://github.com/user-attachments/assets/c0eff690-82eb-4dc2-addf-8559fe367e8a

sejas commented:
@epeicher , maybe we should revert https://github.com/Automattic/studio/pull/1332 ? I don't see much improvement in the progress bar, and it seems it produces other issues.

epeicher committed:
Clear the Export state when completing the Sync

epeicher commented:
> @epeicher , maybe we should revert #1332 ? I don't see much improvement in the progress bar, and it seems it produces other issues.

I have addressed the edge case [reported here](https://github.com/Automattic/studio/pull/1345#pullrequestreview-2822352448) , it was caused by the state of the Export not being reset so [this callback](https://github.com/Automattic/studio/blob/ef98a9a28848ca00cd94c98e3b934573739f59e0/src/hooks/use-import-export.tsx#L259) was returning early. The changes are similar to the [`clearImportState` here](https://github.com/Automattic/studio/blob/ef98a9a28848ca00cd94c98e3b934573739f59e0/src/hooks/sync-sites/use-sync-pull.ts#L215). The commit is ef98a9a.

Given said that. Reverting is always an option. I would continue improving as part of a follow-up task. But I trust your instincts here 😄 (as they are not instinct but knowledge 😉) , if you think there could be other issues, or you don't think the additional complexity is justified, let's revert it.

UPDATE: I have added a revert PR https://github.com/Automattic/studio/pull/1349 to err on the safe side. I close this PR, but save it for future reference.

epeicher commented:
Closing this as the revert https://github.com/Automattic/studio/pull/1349 has been merged. These changes can be used as a reference if the export states need to be emitted for the Sync Push functionality.
